connman/src/connection.c

```diff
@@ -480,19 +480,24 @@ static void set_default_gateway(struct gateway_data *data,
 	if (do_ipv4 && data->ipv4_gateway &&
 			g_strcmp0(data->ipv4_gateway->gateway,
 							"0.0.0.0") == 0) {
-		if (connman_inet_set_gateway_interface(data->index4) < 0)
+		if (connman_inet_set_gateway_interface(data->index4) < 0) {
 			DBG("cannot set v4 gateway interface index %d",
 								data->index4);
+			return;
+		}
+
 		data->ipv4_gateway->active = true;
 		goto done;
```

Comment:
out of curiousity: is there some deeper meaning between duplicated "indicate_default() + mid-function-return" code used above and this "goto where we have common indicate_default() + return" used elsewhere?

Reply:
in any case: there is no behavioral difference, so this is not really relevant for review -> resolving

Reply:
Did some additional restructuring after going through the code to make sure that the active is not anywhere needed in such cases. Made a consistency change to the code: 1abe9b6369c1f4d37223c4eb8b31249eb46c9874

Maybe it just removes some noise from the logs as the unset function is not triggered for an interface that is not actually set here.